game/armedforces/forcegroup.py

```diff
@@ -0,0 +1,250 @@
+from __future__ import annotations
+
+import logging
+import random
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import ClassVar, TYPE_CHECKING, Type, Any, Iterator, Optional
+
+import yaml
+from dcs import Point
+
+from game import db
+from game.data.groups import GroupRole, GroupTask
+from game.data.radar_db import UNITS_WITH_RADAR
+from game.dcs.groundunittype import GroundUnitType
+from game.dcs.shipunittype import ShipUnitType
+from game.dcs.unittype import UnitType
+from game.point_with_heading import PointWithHeading
+from game.layout.layout import TheaterLayout, AntiAirLayout, GroupLayout
+from dcs.unittype import UnitType as DcsUnitType, VehicleType, ShipType, StaticType
+
+from game.theater.theatergroup import TheaterGroup
+
+if TYPE_CHECKING:
+    from game import Game
+    from game.factions.faction import Faction
+    from game.theater import TheaterGroundObject, ControlPoint
+
+
+@dataclass
+class ForceGroup:
+    """A logical group of multiple units and layouts which have a specific tasking"""
+
+    name: str
+    units: list[UnitType[Any]]
+    statics: list[Type[DcsUnitType]]
+    role: GroupRole
+    tasks: list[GroupTask] = field(default_factory=list)
+    layouts: list[TheaterLayout] = field(default_factory=list)
+
+    _by_name: ClassVar[dict[str, ForceGroup]] = {}
+    _by_role: ClassVar[dict[GroupRole, list[ForceGroup]]] = {}
+    _loaded: bool = False
+
+    @staticmethod
+    def for_layout(layout: TheaterLayout, faction: Faction) -> ForceGroup:
+        """TODO Documentation"""
+        units: set[UnitType[Any]] = set()
+        statics: set[Type[DcsUnitType]] = set()
+        for group in layout.groups:
+            for unit_type in group.possible_types_for_faction(faction):
+                if issubclass(unit_type, VehicleType):
+                    units.add(next(GroundUnitType.for_dcs_type(unit_type)))
```

Comment:
This pattern is my fault... but would you mind adding a `UnitType.default_for_dcs_type()` method so we can stop doing this? It confuses me every time I see it, and I'm the one that created the problem so if it's surprising to me it's going to be even worse for others.